src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -210,6 +237,14 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
                 }
             }
 
+            // If the null check is negative and we're not in a while-statement,
+            // the pattern variable is definitely assigned in the subsequent statements.
+            var isWhile = targetStatement.Kind() == SyntaxKind.WhileStatement;
```

Comment:
❓ To help me get my head around this, can you provide an example of why this block is necessary?

Reply:
```cs
if (!(x is Foo  foo))  {} 
// foo is always definitely assigned (we don't need to check so we return early)
while (!(x is Foo  foo)) {}
// foo is always not definitely assigned (so we need to check if it's accessed)
```

Reply:
can you put this in as a comment.

Reply:
Wait, this is only correct if `if` has an unreachable endpoint. I'll remove this check.